.taskcluster.yml

```diff
@@ -6,185 +6,173 @@ version: 1
 policy:
   pullRequests: public
 tasks:
-####################################################################################################
-# Task: Pull requests
-####################################################################################################
-  - $if: 'tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "edited", "synchronize"]'
-    then:
-      $let:
-        decision_task_id: {$eval: as_slugid("decision_task")}
-        expires_in: {$fromNow: '1 year'}
-        scheduler_id: taskcluster-github
-        user: ${event.sender.login}
-        head_branch: ${event.pull_request.head.ref}
-        head_rev: ${event.pull_request.head.sha}
-        repository: ${event.pull_request.head.repo.clone_url}
-        pull_request_title: ${event.pull_request.title}
-        pull_request_number: ${event.pull_request.number}
-        pull_request_url: ${event.pull_request.html_url}
-      in:
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    scheduler_id: taskcluster-github
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.ref}
+        else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    worker_type:
+      $if: 'tasks_for in ["github-push", "github-release"] && event.repository.html_url[:34] == "https://github.com/mozilla-mobile/"'
+      then: mobile-3-decision
+      else: mobile-1-decision
+
+    does_event_occur_on_main_repo:
+      $if: 'event.repository.html_url[:34] == "https://github.com/mozilla-mobile/"'
```

Comment:
Since this `.taskcluster.yml` is specific to `android-components`, I'm not sure that there's a benefit to making this more generic (arguably copy-and-paste-ability, but if we're writing code with the assumption that it's going to be copied-and-pasted, maybe it'd be better to set up a shared abstraction)

The downsides to this being more generic are readability, and the additional brainwork required to figure out what's going on here, e.g.:
* "why the number 34"?
* "does this `.taskcluster.yml` handle more than just `android-components`, or why is it generic here?"

Reply:
Good point. I'll change this :)

Reply:
Good point. I'll change this :)